### PR TITLE
bugfix/support nested iterables

### PR DESCRIPTION
**Pull request by sburba**

### CODE REVIEW

File: api/tests/helpers.py [hunk 1]
```diff
@@ -4,6 +4,14 @@
 _SENTINEL = object()
 
 
+def _is_iterable(instance: Any):
```
Comment (sjohnson31) [inline api/tests/helpers.py:7]: Why did you go with this instead of `isinstance(object, Iterable)` ?

Comment (sburba) [reply api/tests/helpers.py:7]: The official docs recommend against it, because it misses some edge cases.

> Checking isinstance(obj, Iterable) detects classes that are registered as Iterable or that have an __iter__() method, but it does not detect classes that iterate with the __getitem__() method. The only reliable way to determine whether an object is iterable is to call iter(obj).

https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable

Comment (sjohnson31) [reply api/tests/helpers.py:7]: All right.

**Review by sjohnson31 — APPROVED** [review]
